src/main/java/tr/com/srdc/cda2fhir/transform/CCDTransformerImpl.java

```diff
@@ -197,23 +214,29 @@ public Bundle transformDocument(ClinicalDocument cda) {
         }
 
         // init the global ccd bundle via a call to resource transformer, which handles cda header data (in fact, all except the sections)
-        Bundle ccdBundle = resTransformer.tClinicalDocument2Composition(ccd);
+        Bundle ccdBundle = resTransformer.tClinicalDocument2Bundle(ccd, includeComposition);
+        
         // the first bundle entry is always the composition
-        Composition ccdComposition = (Composition)ccdBundle.getEntry().get(0).getResource();
+        Composition ccdComposition = includeComposition ? (Composition)ccdBundle.getEntry().get(0).getResource() : null;
+        
         // init the patient id reference if it is not given externally. the patient is always the 2nd bundle entry
-        if (patientRef == null)
-            patientRef = new Reference(ccdBundle.getEntry().get(1).getResource().getId());
-        else // Correct the subject at composition with given patient reference.
+        if (patientRef == null) {
+            patientRef = new Reference(ccdBundle.getEntry().get(1).getResource().getId()); // TO DO: Do not depend on the order
+        } else if (ccdComposition != null) { // Correct the subject at composition with given patient reference.
```

Comment:
Just a general note that would have applied before this PR. I don't see the partientRef being set anywhere in tClinicalDocument2Bundle, so the conditional: ``` if(patientRef == null) {...} else { ccdComposition.setSubject(patientRef)}``` confuses me. 

The subject could have already been set in addition to the patient ref  also being null in the transformDocument function. 

The logic for the subject should be the same within tClinicalDocument2Bundle and transformDocument, correct? If so, why is it being set twice? It seems like it will always be overwritten.

Reply:
Right. patientRef they are doing has some problems. We should not be setting patientRef member variable in an unrelated method to starters; that is just a bad coding practice. We need to look into this PatientRef more closely based on discussions with them. It just does not belong to this PR.